### PR TITLE
[WIP] Upgrade to prompt_toolkit 2.0.

### PR DESCRIPTION
**Pull request by jonathanslenders**
## Description
Upgrade to prompt_toolkit 2.0.
Work in progress - needs more testing.

## Checklist

- [ ] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by tsroten** [comment]
@jonathanslenders Nice work so far! Thanks for spending the time to get mycli updated. I'll check this out and do some testing as well.

File: mycli/filters.py [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 
-class HasSelectedCompletion(Filter):
-    """Enable when the current buffer has a selected completion."""
 
-    def __call__(self, cli):
-        complete_state = cli.current_buffer.complete_state
-        return (complete_state is not None and
-                 complete_state.current_completion is not None)
+@Condition
+def has_selected_completion():
```
Comment (meeuw) [inline mycli/filters.py]: is this obsoleted by `prompt_toolkit.filters.completion_is_selected`?

**Comment by thorstenkampe** [comment]
prompt_toolkit 2.0 is now pulled in by pgcli and IPython which in turn results in mycli failing to start.

**Comment by meeuw** [comment]
@thorstenkampe thanks for your reminder, I've ported the pgcli prompt_toolkit changes to mycli in #633 could you please test this branch using: `pip install git+http://github.com/dbcli/mycli@feature/prompt-toolkit-2.0 --upgrade`

**Comment by thorstenkampe** [comment]
@meeuw I'm confirming that the branched mycli works with prompt_toolkit 2.0.

**Comment by meeuw** [comment]
has been merged

**Pull request closed without merging**